CoreScriptsRoot/Modules/Server/ClientChat/MessageLabelCreator.lua

```diff
@@ -77,23 +93,83 @@ local function WrapIntoMessageObject(id, BaseFrame, BaseMessage, Tweener, Strong
 	end
 
 	function obj:Destroy()
-		self.BaseFrame:Destroy()
-		self.BaseMessage:Destroy()
+		ReturnToObjectPoolRecursive(self.BaseFrame, self.ObjectPool)
 	end
 
 	return obj
 end
 
+function methods:CreateMessageLabelFromType(messageData, messageType)
+	return self["Create" ..messageType.. "Label"](self, messageData)
+end
 
+function methods:CreateBaseMessage(message, font, fontSize, chatColor)
+	local BaseFrame = self.ObjectPool:GetInstance("Frame")
```

Comment:
This will be part of the "Hook for player customize message label system".
My thought: apply default styles first, the pass the label through the list of user messagelabel plugins.